bin/sozo/src/commands/migrate.rs

```diff
@@ -147,7 +150,19 @@ pub async fn setup_env<'a>(
         let provider = starknet.provider(env)?;
         trace!(?provider, "Provider initialized.");
 
-        let spec_version = provider.spec_version().await?;
```

Comment:
What's the current timeout by default?

Reply:
forever

Reply:
hmmm would be better if we can introduce this timeout in the `Provider` implementation itself. so that it'd be applied to all request.

i assume the current approach is to just add the timeout on the very first provider request for a command.

Reply:
Agree, would be here @notV4l: https://github.com/dojoengine/dojo/blob/21ec45eb94f8760333919c2916cf740f5d6bdb1a/bin/sozo/src/commands/options/starknet.rs#L27. Creating a client with the builder and timeout instead.